agents/trainer/task.py

```diff
@@ -0,0 +1,319 @@
+                     "The number of steps.")
+
+# Network
+flags.DEFINE_string("network", "agents.scripts.networks.feed_forward_gaussian",
+                    "The registered network name to use for policy and value.")
+# TODO: Make it possible to pass policy and value layers via hparam config.
+# flags.DEFINE_string("policy_layers", "200,100",
+#                      "A comma-separates list of the size of a series of layers "
+#                      "to comprise the policy network.")
+# flags.DEFINE_string("value_layers", "200,100",
+#                      "A comma-separates list of the size of a series of layers "
+#                      "to comprise the value network.")
+flags.DEFINE_float("init_mean_factor", 0.1,
+                   "")
+flags.DEFINE_float("init_std", 0.35,
+                   "")
+
+# Optimization
+flags.DEFINE_float("learning_rate", 1e-4,
+                   "The learning rate of the optimizer.")
+flags.DEFINE_string("optimizer", "tensorflow.train.AdamOptimizer",
+                    "The import path of the optimizer to use.")
+flags.DEFINE_integer("update_epochs", 25,
+                     "The number of update epochs.")
+flags.DEFINE_integer("update_every", 60,
+                     "The update frequency.")
+# flags.DEFINE_boolean("optimizer_pre_initialize", True,
+#                      "Whether to pre-initialize the optimizer.")
+
+# Losses
+flags.DEFINE_float("discount", 0.995,
+                   "The discount.")
+flags.DEFINE_float("kl_target", 1e-2,
+                   "the KL target.")
+flags.DEFINE_integer("kl_cutoff_factor", 2,
+                     "The KL cutoff factor.")
+flags.DEFINE_integer("kl_cutoff_coef", 1000,
+                     "The KL cutoff coefficient.")
+flags.DEFINE_integer("kl_init_penalty", 1,
+                     "The initial KL penalty?.")
+
+FLAGS = flags.FLAGS
+
+
+def hparams_base():
+  """Base hparams tf/Agents PPO """
+
+  # General
+  algorithm = agents.ppo.PPOAlgorithm
+  num_agents = 30
+  eval_episodes = 30
+  use_gpu = False
+
+  # Environment
+  env = 'KukaBulletEnv-v0'
+  normalize_ranges = True
+  max_length = 1000
+
+  # Network
+  network = agents.scripts.networks.feed_forward_gaussian
+  weight_summaries = dict(
+      all=r'.*', policy=r'.*/policy/.*', value=r'.*/value/.*')
+  policy_layers = 200, 100
+  value_layers = 200, 100
+  init_output_factor = 0.1
+  init_logstd = -1
+  init_std = 0.35
+
+  # Optimization
+  update_every = 60
+  update_epochs = 25
+  optimizer = tf.train.AdamOptimizer
+  learning_rate = 1e-4
+  steps = 3e7  # 30M
+
+  # Losses
+  discount = 0.995
+  kl_target = 1e-2
+  kl_cutoff_factor = 2
+  kl_cutoff_coef = 1000
+  kl_init_penalty = 1
+
+  return locals()
+
+
+def _object_import_from_string(name):
+
+  components = name.split('.')
+  mod = __import__(components[0])
+  for comp in components[1:]:
+    mod = getattr(mod, comp)
+  return mod
+
+
+def _realize_import_attrs(d, filter):
+  for k, v in d.items():
+    if k in filter:
+      # try:
```

Comment:
Remove unused debugging code.

Reply:
Good thought about removing unused code I'll take a look for that.

But that's not debugging code, it's used to allow people to set the algorithm, network, and optimizer - all which need to be python objects - from the level of Ksonnet. Which in general I think is a convenient feature for being able to also compare e.g. feedforward to recurrent or in the future the performance of different algorithms. Check out the [demo notebook](https://github.com/cwbeitel/examples/blob/master/agents/demo/demo.ipynb) where jobs are being created and you're able to specify the algorithm, optimizer, etc.

But this is not the way I would implement it if doing it right this was a little hack I should have caught on a read-through. It accesses FLAGS using an internal method which not surprisingly breaks in 1.5. So I should probably fix that.

Of course there are other ways to accomplish this such as with the registry pattern used in tensor2tensor, e.g. [this](https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/models/transformer.py#L47) and [this](https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/models/transformer.py#L874).

Reply:
I was mainly referring to the commented out code, i.e. the try-catch block here. I'm not sure if that's what you were talking about but is there a reason that should be checked in?

Reply:
Oh good point thank you.